CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/ExportToSVGAction.java

```diff
@@ -0,0 +1,140 @@
+import au.gov.asd.tac.constellation.graph.node.GraphNode;
+import au.gov.asd.tac.constellation.graph.schema.visual.attribute.objects.ConnectionMode;
+import au.gov.asd.tac.constellation.graph.schema.visual.concept.VisualConcept;
+import au.gov.asd.tac.constellation.plugins.PluginExecution;
+import au.gov.asd.tac.constellation.plugins.importexport.ImportExportPluginRegistry;
+import au.gov.asd.tac.constellation.utilities.file.FileExtensionConstants;
+import au.gov.asd.tac.constellation.utilities.gui.filechooser.FileChooser;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.File;
+import javax.swing.filechooser.FileFilter;
+import org.apache.commons.lang3.StringUtils;
+import org.openide.DialogDisplayer;
+import org.openide.NotifyDescriptor;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionRegistration;
+import org.openide.filesystems.FileChooserBuilder;
+import org.openide.filesystems.FileChooserBuilder.SelectionApprover;
+import org.openide.util.NbBundle;
+
+/**
+ * Action responsible for triggering <code>ExportToSVGPlugin</code> that exports Graphs to SVG files.
+ * 
+ * @author capricornunicorn123
+ */
+@ActionID(category = "File", id = "au.gov.asd.tac.constellation.plugins.importexport.image.ExportToSVG")
+@ActionRegistration(displayName = "#CTL_ExportToSVG",
+        iconBase = "au/gov/asd/tac/constellation/plugins/importexport/image/exportToImage.png", // 
+        surviveFocusChange = true)
+@ActionReference(path = "Menu/File/Export", position = 50)
+@NbBundle.Messages("CTL_ExportToSVG=To SVG")
+public final class ExportToSVGAction implements ActionListener {
+
+    private static final String TITLE = "Export to SVG";
+    
+    private final GraphNode context; 
+    
+    public ExportToSVGAction(final GraphNode context) {
+        this.context = context;
+    }
+   
+    @Override
+    public void actionPerformed(final ActionEvent e) {
+        final ReadableGraph graph = context.getGraph().getReadableGraph();
+        
+        //The graph has data on it so it can be exported
+        if (graph.getVertexCount() != 0){        
+            FileChooser.openSaveDialog(getExportToSVGFileChooser()).thenAccept(optionalFile -> optionalFile.ifPresent(file -> {
+                String fnam = file.getAbsolutePath();
+
+                if (!fnam.toLowerCase().endsWith(FileExtensionConstants.SVG)) {
+                    fnam += FileExtensionConstants.SVG;
+                }
+
+                int connectionModeID = VisualConcept.GraphAttribute.CONNECTION_MODE.get(graph);
+                ConnectionMode currentConnectionmode = graph.getObjectValue(connectionModeID, 0);
+                
+                PluginExecution.withPlugin(ImportExportPluginRegistry.EXPORT_SVG)
+                        .withParameter(ExportToSVGPlugin.FILE_NAME_PARAMETER_ID, fnam)
+                        .withParameter(ExportToSVGPlugin.GRAPH_TITLE_PARAMETER_ID, "Milestone 4")
```

Comment:
I'm just assuming this will be updated by the end but just bringing to your attention if didn't already know